bin/install

```diff
@@ -1,158 +1,502 @@
+  set -g OMF_CONFIG "$OMF_CONFIG_DEFAULT"
+
+  # Ensure repository URL ends with .git
+  set OMF_REPO_URI (echo $OMF_REPO_URI | command sed 's/\.git//').git
+
+  # Parse command args
+  while set -q argv[1]
+    switch "$argv[1]"
+      case --help -h '/?'
+        echo "Usage: install [options]
+  Install Oh My Fish
+
+Options:
+  --config=<path>         Put config in a specific path (default is $OMF_CONFIG_DEFAULT).
+  --help, -h              Show this help message.
+  --noninteractive        Disable interactive questions (assume no, use with --yes to assume yes).
+  --offline[=<path>]      Offline install, optionally specifying a tar or directory to use.
+  --path=<path>           Use a specific install path (default is $OMF_PATH_DEFAULT).
+  --uninstall             Uninstall existing installation instead of installing.
+  --yes, -y               Assume yes for interactive questions.
+"
+        return 0
+
+      case '--config=*'
+        echo "$argv[1]" | command cut -d= -f2 | read -g OMF_CONFIG
+
+      case --noninteractive
+        set -g NONINTERACTIVE
+
+      case --offline
+        set -g OFFLINE
+
+      case '--offline=*'
+        set -g OFFLINE
+        echo "$argv[1]" | command cut -d= -f2 | read -g OFFLINE_PATH
+
+      case '--path=*'
+        echo "$argv[1]" | command cut -d= -f2 | read -g OMF_PATH
+
+      case --uninstall
+        set -g UNINSTALL
+
+      case --yes -y
+        set -g ASSUME_YES
+
+      case '*'
+        abort "Unrecognized option '$argv[1]'. Try 'install --help' for usage."
+    end
+    set -e argv[1]
+  end
 
-function report -a what message
-  switch $what
-  case 'progress'
-    set_color yellow
-  case 'success'
-    set_color green
-  case 'error'
-    set_color red
-    printf "$message\n"
-    exit 1
+  # Ensure the environment meets all of the requirements.
+  assert_cmds
+  assert_fish_version_compatible 2.2.0
+  assert_git_version_compatible 1.9.5
+  assert_interactive
+
+  # If the user wants to uninstall, jump to uninstallation and exit.
+  if set -q UNINSTALL
+    uninstall_omf
+    return
   end
 
-  printf "$message\n"
+  # Check if OMF is already installed.
+  if test -d "$OMF_PATH"
+    say "Existing installation detected at $OMF_PATH"
+
+    confirm_yes "Would you like to remove the existing installation?"
+    uninstall_omf
+  end
+
+  # Begin the install process.
+  install_omf
+
+  # We made it!
+  say "Installation successful!"
+
+  # Open a brand new shell if we are in interactive mode.
+  set -q NONINTERACTIVE
+    or exec fish < /dev/tty
+
+  return 0
 end
 
-function fish_version_compatible
-  set -q FISH_VERSION; or set -l FISH_VERSION $version
-  set -l major (echo $FISH_VERSION | cut -d. -f1)
-  set -l minor (echo $FISH_VERSION | cut -d. -f2)
 
-  return (test $major = $OMF_FISH_MIN_VER[1] -a $minor -ge $OMF_FISH_MIN_VER[2])
+# Add an exit hook to display a message if the installer aborts or errors.
+function on_exit -p %self
+  if test $status -ne 0 -a $status -ne 2
```

Comment:
`$status` is indeed usable here since the process I'm handling the exit of is myself, so `$status` is still the most recent exit status. `$argv` is probably more correct though. I shall change it.